test/model/message_list_test.dart

```diff
@@ -544,65 +509,76 @@ void main() {
-          .where((m) => m.id != 150).toList());
-      model.notifyListenersIfMessagePresent(150);
-      checkNotNotified();
-    });
+    test('not in narrow', () async {
+      final stream = eg.stream();
+      await prepare(narrow: ChannelNarrow(stream.streamId));
+      await prepareMessages(foundOldest: true, messages:
+        List.generate(30, (i) => eg.streamMessage(stream: stream)));
 
-    test('message absent (older than window)', () async {
-      await prepare(narrow: const CombinedFeedNarrow());
-      await prepareMessages(foundOldest: false,
-        messages: List.generate(100, (i) => eg.streamMessage(id: 100 + i)));
-      model.notifyListenersIfMessagePresent(50);
+      check(model).messages.length.equals(30);
+      final otherStream = eg.stream();
+      await store.handleEvent(MessageEvent(id: 0,
+        message: eg.streamMessage(stream: otherStream)));
       checkNotNotified();
+      check(model).messages.length.equals(30);
     });
 
-    test('message absent (newer than window)', () async {
-      await prepare(narrow: const CombinedFeedNarrow());
-      await prepareMessages(foundOldest: false,
-        messages: List.generate(100, (i) => eg.streamMessage(id: 100 + i)));
-      model.notifyListenersIfMessagePresent(250);
+    test('before fetch', () async {
+      final stream = eg.stream();
+      await prepare(narrow: ChannelNarrow(stream.streamId));
+      await store.handleEvent(MessageEvent(id: 0,
+        message: eg.streamMessage(stream: stream)));
       checkNotNotified();
+      check(model).fetched.isFalse();
     });
-  });
 
-  group('notifyListenersIfAnyMessagePresent', () {
-    final messages = List.generate(100, (i) => eg.streamMessage(id: 100 + i));
+    test('parse ZulipContent', () async {
+      final stream = eg.stream();
+      await prepare(narrow: ChannelNarrow(stream.streamId));
+      await prepareMessages(foundOldest: true, messages: []);
 
-    test('all messages present', () async {
-      await prepare(narrow: const CombinedFeedNarrow());
-      await prepareMessages(foundOldest: false, messages: messages);
-      model.notifyListenersIfAnyMessagePresent([150, 151, 152]);
+      await store.handleEvent(MessageEvent(id: 0,
+        message: eg.streamMessage(stream: stream)));
+      // Each [checkNotifiedOnce] call ensures there's been a [checkInvariants]
+      // call, where the [ContentNode] gets checked.  The additional checks to
+      // make this test explicit.
       checkNotifiedOnce();
+      check(model).messages.single.poll.isNull();
+      check(model).contents.single.isA<ZulipContent>();
     });
 
-    test('some messages present', () async {
-      await prepare(narrow: const CombinedFeedNarrow());
-      await prepareMessages(foundOldest: false,
-        messages: messages.where((m) => m.id != 151).toList());
-      model.notifyListenersIfAnyMessagePresent([150, 151, 152]);
+    test('parse PollContent', () async {
```

Comment:
These two new cases don't belong in a group that's about `handleMessageEvent`, because they're not really about that method — in particular they're not any more specific to that method than they are to `fetchInitial` or `fetchOlder`.

In general this test file is one where our usual principle of organizing the tests in the same order as the code they test often breaks down: `model/message_list.dart` is one of our most intricate areas of model code, and might be the one file with the greatest complexity of interrelationships between different parts of its implementation.

You can see that organizing principle already broken down in roughly the latter half of the file: the "regression tests for …" and "stream/topic muting" groups, the tests "showSender is maintained correctly" and "recipient headers are maintained consistently", and the `checkInvariants` function which underlies most of the tests and describes invariants that cut across the implementation code.

So these can go in a new small group of their own, perhaps just above "recipient headers are maintained consistently" since they're both about logic within _MessageSequence.

Then I think we don't need these commits:
1919c1ba2 msglist test [nfc]: Make a group for MessageEvent tests
99a62b71a msglist test [nfc]: Name test groups after the event handler method
b69f0bfef msglist test [nfc]: Move notifyListeners tests to match implementation